### DISCUSSION TITLE
Autogen handoff

### DISCUSSION THREAD

**Discussion by tinased95** [microsoft/autogen#3944] [Q&A]
Hi, I’m using the Handoff design pattern example, and it works well overall. However, when I ask a sales-related question, the task is successfully delegated to the SalesAgent, and the conversation proceeds smoothly. But when I ask something irrelevant to SalesAgent, it says it’s delegating back to triage, then fails with this error:

`openai.InternalServerError: Error code: 500 - {'error': {'message': 'The model produced invalid content. Consider modifying your prompt if you are seeing this error persistently.', 'type': 'model_error', 'param': None, 'code': None}}`

In short, triage can delegate tasks to the sub agents but when sub agents want to delegate task to triage again it fails

**Comment by ekzhu** [comment]
This is a 500 internal sever error we have started to see recently from open ai endpoints. Which model are you using? 

It's not something we can fix at this point as it comes from open ai.

**Reply by tinased95** [reply]
I've use GPT-4o and GPT-4o-mini. Both have this issue.

**Reply by ekzhu** [reply]
I just tried with OpenAI endpoint, and I couldn't reproduce it. It worked for me using both models you tried. Have you tried multiple times?

**Reply by tinased95** [reply]
Yes, but I haven’t had any luck.

Could you please try the following requests?

1. I want a refund
2. A pair of shoes I bought
3. no sorry I changed my mind, I want to buy a new pair of shoes (this is to force the IssuesAndRepairsAgent to delegate the task back to Triage)

After that, I received the following response:

```
Delegating to TriageAgent
Error processing publish messageopenai.InternalServerError: Error code: 500 - {'error': {'message': 'The model produced invalid content. Consider modifying your prompt if you are seeing this error persistently.', 'type': 'model_error', 'param': None, 'code': None}}
```


It seems that Triage can delegate tasks to the IssuesAndRepairsAgent, but there are errors when trying to do it in the opposite direction.

**Reply by ekzhu** [reply]
```
--------------------------------------------------------------------------------
User login, session ID: 327266a7-00fb-41a7-99d8-c316bdab3951.
--------------------------------------------------------------------------------
User:
I want a refund
--------------------------------------------------------------------------------
TriageAgent:
I can assist you with that! Could you tell me if this is related to an issue with a product or something else?
--------------------------------------------------------------------------------
User:
A pair of shoes
--------------------------------------------------------------------------------
TriageAgent:
[FunctionCall(id='call_vAw2typMzTfgKYmrMs8xeCeb', arguments='{}', name='transfer_to_issues_and_repairs')]
--------------------------------------------------------------------------------
TriageAgent:
Delegating to IssuesAndRepairsAgent
--------------------------------------------------------------------------------
IssuesAndRepairsAgent:
I'm here to help! What seems to be the issue with the pair of shoes?
--------------------------------------------------------------------------------
User:
no I changed my mind bring me back 
--------------------------------------------------------------------------------
IssuesAndRepairsAgent:
[FunctionCall(id='call_lzd4bxj1MDurXBTYgSXQX8II', arguments='{}', name='transfer_back_to_triage')]
--------------------------------------------------------------------------------
IssuesAndRepairsAgent:
Delegating to TriageAgent
--------------------------------------------------------------------------------
TriageAgent:
[FunctionCall(id='call_gO8PXrDlEApydcmeqLyDJmg2', arguments='{}', name='escalate_to_human')]
--------------------------------------------------------------------------------
TriageAgent:
Delegating to HumanAgent
```

Works fine for me. gpt-4o-mini

**Reply by ekzhu** [reply]
What's your `autogen-core` package version?

**Reply by tinased95** [reply]
autogen-core: 0.4.0.dev2

**Reply by tinased95** [reply]
The only change I added to the notebook is to use `AzureOpenAIChatCompletionClient` instead of `OpenAIChatCompletionClient`

**Reply by tinased95** [reply]
I'm using this setting for `AzureOpenAIChatCompletionClient`

```
model_client = AzureOpenAIChatCompletionClient(
    model="test-gpt4o",
    api_version="2023-03-15-preview",
    azure_endpoint=azure_endpoint,
    azure_ad_token_provider=token_provider,
    model_capabilities={
        "vision":False,
        "function_calling":True,
        "json_output":True,
    }
)
```

**Reply by JComyn** [reply]
Hi, did you find any solution to this problem? I'm facing a very similar error and also using AzureOpenAIChatCompletionClient.

**Reply by ekzhu** [reply]
Can you use a more recent gpt-4o version?

**Reply by JComyn** [reply]
Using other models, like gpt-4, solved the issue.

**Reply by xrl0123** [reply]
Hi, I'm still experiencing this issue. I've tried gpt-4o 2024-11-20 and gpt-4o 2024-08-06 so far. Any recommendations for the model version I should use? Thanks!
